packages/client/src/raffles/types/postRaffleParticipation.types.ts

```diff
@@ -0,0 +1,23 @@
+import type { Config } from '../..';
+import type {
+  Raffle,
+  RaffleParticipation,
+  RaffleParticipationStatus,
+} from './raffles.types';
+
+export type PostParticipationData = {
+  userId: string;
```

Comment:
```suggestion
  userId: RaffleParticipation['userId'];
```

It seems that all of these values can come from the `RaffleParticipation`, what do you think?

Reply:
done

Reply:
I meant all types:

```ts
billingAddressId: RaffleParticipation['billingAddressId'];
shippingAddressId: RaffleParticipation['shippingAddressId'];
```

and so on.

You can also try another thing, like using [Pick](https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys), for example:

```ts
export type PostParticipationData = Pick<RaffleParticipation, 'userId' | 'billingAddressId' | ...>
```

Or either [Omit](https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys) if you think it's a better fit.

Leaving the suggestion 🙂

Reply:
Nice! I'm going to use pick to start getting more familiar with this